### ISSUE TITLE
Tracking issue for `use $crate;` compatibility warning

### ISSUE THREAD

**Issue by jseyfried** [rust-lang/rust#37390]
The warning was introduced in #37361 to avoid breakage from #37213.

**Comment by brson** [comment]
Presumably the next step is to make this an error.

**Comment by steveklabnik** [comment]
Triage: Digging through all the linked issues is tough, and it's not clear to me exactly what this warning is or does. @jseyfried can you maybe elaborate a little?

**Comment by petrochenkov** [comment]
This needs to be turned into an error, i.e. behave like `use crate;` basically:
```rust
fn main() {
    use crate; // ERROR crate root imports need to be explicitly named: `use crate as name;`
}
```

**Comment by Centril** [comment]
Relevant code is here:
https://github.com/rust-lang/rust/blob/488381ce9ef0ceabe83b73127c659e5d38137df0/src/librustc_resolve/build_reduced_graph.rs#L467

Should be easy to turn that into `.struct_span_err` and then do a crater run.
